### PR TITLE
Support single object tool_calls in OpenAI vendor APIs

### PR DESCRIPTION
**Pull request by liugddx**
Thank you for taking time to contribute this pull request!
You might have already read the [contributor guide][1], but as a reminder, please make sure to:

* Sign the [contributor license agreement](https://cla.pivotal.io/sign/spring)
* Rebase your changes on the latest `main` branch and squash your commits
* Add/Update unit tests as needed
* Run a build and make sure all tests pass prior to submission

### CODE REVIEW

**Comment by liugddx** [comment]
I am using a commercial LLM, which is based on the openai interface specification, but the difference is that its function call is slightly different. Its tool_calls is an object not an array.  https://www.xfyun.cn/doc/spark/HTTP%E8%B0%83%E7%94%A8%E6%96%87%E6%A1%A3.html#_3-%E8%AF%B7%E6%B1%82%E8%AF%B4%E6%98%8E

**Comment by liugddx** [comment]
I am using a commercial LLM, which is based on the openai interface specification, but the difference is that its function call is slightly different. Its tool_calls is an object not an array.  https://www.xfyun.cn/doc/spark/HTTP%E8%B0%83%E7%94%A8%E6%96%87%E6%A1%A3.html#_3-%E8%AF%B7%E6%B1%82%E8%AF%B4%E6%98%8E

**Comment by liugddx** [comment]
I am using a commercial LLM, which is based on the openai interface specification, but the difference is that its function call is slightly different. Its tool_calls is an object not an array.  https://www.xfyun.cn/doc/spark/HTTP%E8%B0%83%E7%94%A8%E6%96%87%E6%A1%A3.html#_3-%E8%AF%B7%E6%B1%82%E8%AF%B4%E6%98%8E

File: models/spring-ai-openai/src/main/java/org/springframework/ai/openai/api/OpenAiApi.java [hunk 1]
```diff
@@ -971,7 +972,8 @@ public record ChatCompletionMessage(// @formatter:off
 			@JsonProperty("role") Role role,
 			@JsonProperty("name") String name,
 			@JsonProperty("tool_call_id") String toolCallId,
-			@JsonProperty("tool_calls") List<ToolCall> toolCalls,
+			@JsonProperty("tool_calls")
+			@JsonFormat(with = JsonFormat.Feature.ACCEPT_SINGLE_VALUE_AS_ARRAY)List<ToolCall> toolCalls,
```
Comment (jitokim) [inline models/spring-ai-openai/src/main/java/org/springframework/ai/openai/api/OpenAiApi.java:976]:
```suggestion
			@JsonFormat(with = JsonFormat.Feature.ACCEPT_SINGLE_VALUE_AS_ARRAY) List<ToolCall> toolCalls,
```

How about adding spaces like this?

Comment (liugddx) [reply models/spring-ai-openai/src/main/java/org/springframework/ai/openai/api/OpenAiApi.java]: thanks for the reminder.

**Comment by liugddx** [comment]
@tzolov PTAL

**Comment by markpollack** [comment]
looks more like a tweak to be more broadly compatable but not related to fault tolerance.  Change seems like it won't cause any side effects, so i'll merge after tests run green.

**Comment by markpollack** [comment]
merged in f92450279148cfe9988b5224c3bb00da55dcbfb7

**Pull request closed without merging**